### ISSUE TITLE
Show comments doesn't work on Firefox with Dark Reader active

### ISSUE THREAD

**Issue by thjderjktyrjkt** [RickyRomero/shut-up-webextension#17]
As title says. Toggle to show comments won't work, even after refreshing the page.
The only way to make them reappear is to disable this extension from extension settings.
Chromium browsers are unaffected.

**Comment by GoldenHashtag** [comment]
Never mind. I've switched to [shutup.css](https://github.com/panicsteve/shutup-css) with [Stylus](https://add0n.com/stylus.html) (No Stylish because of [privacy concern](https://robertheaton.com/2018/07/02/stylish-browser-extension-steals-your-internet-history/)).
You may still look into this issue if you want though.

**Comment by RickyRomero** [comment]
I've seen people mention this problem a few times now, but I'm not sure what the cause is. Toggling comments has *always* worked in Firefox for me, without exception!

https://user-images.githubusercontent.com/233891/139966946-2e3c0dd3-c702-4389-9f30-e3df549723eb.mov

If anyone stumbling upon this GitHub issue looking for help sees this post because they're also seeing this behavior, could you confirm a few things with me?

1. Does this behavior only occur on specific websites, or does it happen everywhere?
1. Do you notice any warnings in the developer console when you try to toggle comments?
1. Is this happening in Private Browsing, or does it happen in normal windows too?
1. Can you think of any nonstandard config options you're using in Firefox that might have side effects?
  As an example: You might not think **about:preferences#privacy > Delete cookies and site data when Firefox is closed** has any effect, but it actually causes *all* windows to be treated like Private Browsing windows.

Basically, I need to be able to see this issue myself in order to fix it... but as of yet, I've never seen it, and I don't have any clues. Hoping someone out there can help.

**Comment by thjderjktyrjkt** [comment]
Can you copy the website and paste it there. I'll try testing it. (Even though currently I have no issue with Stylus turning off comments)

**Comment by thjderjktyrjkt** [comment]
Ok so now that's some plain magic. I've just tried it again and it worked now. Perhaps some Firefox issues that got patch in later version? I'm on 94.0.1 now. But let's keep this issue opened for a while before making a final judgement.

**Comment by thjderjktyrjkt** [comment]
Nope it doesn't work again.
https://news.ycombinator.com/item?id=22752131

It appears to be site-specific. For example this one, showing comment won't work until you refresh it, then it'll work for once. But if you refresh it again then comments are broken again.

**Comment by RickyRomero** [comment]
@thjderjktyrjkt Thank you for testing! This bug is really weird. I recorded this on 94.0.1 too:

https://user-images.githubusercontent.com/233891/141245166-68227970-7fec-4c4e-9d7a-db255add68cf.mp4

🤷⁉️

What happens if you go into `about:addons` and turn off every other extension, then reload the page and try it? Does it work then? The only other extension I have installed is 1Password, and I'm wondering if Shut Up might be conflicting with another extension on your system.

Also... which OS are you running? It should work fine on any OS – even Linux – but maybe we've discovered a new quirk? I'm on macOS but I can test on anything.

**Comment by thjderjktyrjkt** [comment]
`about:profile`, created a new profile solely to test this.

New profile (Only this extension is installed, no tweak applied):

https://user-images.githubusercontent.com/40300931/141246226-c9c5939d-b042-449b-8941-40058e31670c.mp4

My existing profile with tons of extensions and tweaks:

https://user-images.githubusercontent.com/40300931/141246266-5f02df9c-3616-4568-b134-6e978988b713.mp4


I'm using Fedora. But this issue also occurs on Windows, so it shouldn't be OS specific.

**Comment by thjderjktyrjkt** [comment]
So now the only remaining problems are:
 - What caused conflict with shutup extension?
 - Why is Stylus not affected?

**Comment by thjderjktyrjkt** [comment]
It's Dark Reader.

https://user-images.githubusercontent.com/40300931/141247454-c16b83ef-91f8-4269-a3c1-3737640281ac.mp4

But on Chrome Shut Up extension doesn't have issue with Dark Reader. Nor does Stylus, so might need to have further look into this problem.

**Comment by RickyRomero** [comment]
Perfect! I really appreciate you narrowing down the cause. I can confirm the same issue on my Mac.

https://user-images.githubusercontent.com/233891/141250289-4eb596e6-2aa3-4b60-aef4-46b4a4f0dd77.mp4

Now I can try to fix the conflict! :+1:

**Comment by RickyRomero** [comment]
Okay, I've done some investigation and determined that I'll need some help from the Dark Reader team to fully resolve this. Bug: https://github.com/darkreader/darkreader/issues/7347

What's happening is their extension is copying my stylesheet into their own style tag, and then they're re-applying all my styles. So when I remove my copy to show comments, their copy of my stylesheet is still in effect, and the comments remain hidden as a result.

(I noticed another issue where Dark Reader and Shut Up are competing to get their stylesheets to the end of the `<head>` tag, which I'm resolving separately. However, that doesn't appear to be the root cause of this bug.)

**Comment by RickyRomero** [comment]
As of 14e3ac2c1d182968c0f9dd11c30ccbcb47524028 I've addressed the `MutationObserver` conflicts with Dark Reader. That patch to Shut Up will go live across the various extension galleries over the coming hours.

For Dark Reader's part, it looks like they fixed the duplicate styles bug on their end with a [pretty simple change](https://github.com/darkreader/darkreader/commit/6603943e6bacaa4fbcfd99814035b71bbd78f344). They haven't released that fix yet, but given their frequent release schedule it should happen soon.

I'll close this bug after I've verified the fix in Dark Reader.
